Source/Laser/LaserParticleContainer.cpp

```diff
@@ -609,31 +611,40 @@ LaserParticleContainer::PushP (int lev, Real dt,
 void
 LaserParticleContainer::calculate_laser_plane_coordinates (
     const int np, const int thread_num,
-    Real * const pplane_Xp,
-    Real * const pplane_Yp)
+    Real * AMREX_RESTRICT const pplane_Xp,
+    Real * AMREX_RESTRICT const pplane_Yp)
 {
     Real const * const AMREX_RESTRICT xp = m_xp[thread_num].dataPtr();
     Real const * const AMREX_RESTRICT yp = m_yp[thread_num].dataPtr();
     Real const * const AMREX_RESTRICT zp = m_zp[thread_num].dataPtr();
-    Real const * const AMREX_RESTRICT tmp_u_X = u_X.dataPtr();
-    Real const * const AMREX_RESTRICT tmp_u_Y = u_Y.dataPtr();
-    Real const * const AMREX_RESTRICT tmp_position = position.dataPtr();
+    Real tmp_u_X_0 = u_X[0];
```

Comment:
Thanks for this PR @WeiqunZhang.
Why do we need `Real` instead of `Real*` here?

Reply:
The lambda function captures things by value.  In the case of `Real*`, the value of the pointer is captured, but the pointer itself is not a device pointer.  So it didn't work.  It actually worked on summit because any memory allocated through `new` or `malloc` could be accessed on GPU.  But that's the case for x86 machines.

Reply:
I meant not the case for x86.

Reply:
Thanks!